### PR TITLE
chore: refactor ARM logic

### PR DESCRIPTION
**Pull request by artunduman**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Some refactoring, due to the fact that we only build 22.04 and 24.04 ARM64 on Ubuntu

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 1]
```diff
@@ -66,16 +66,7 @@ installDeps() {
 }
 
 updateAptWithMicrosoftPkg() {
-    if [[ $(isARM64) == 1 ]]; then
-        if [[ "${UBUNTU_RELEASE}" == "22.04" || "${UBUNTU_RELEASE}" == "24.04" ]]; then
-            retrycmd_if_failure_no_stats 120 5 25 curl https://packages.microsoft.com/config/ubuntu/${UBUNTU_RELEASE}/prod.list > /tmp/microsoft-prod.list || exit $ERR_MOBY_APT_LIST_TIMEOUT
-        else
-            retrycmd_if_failure_no_stats 120 5 25 curl https://packages.microsoft.com/config/ubuntu/${UBUNTU_RELEASE}/multiarch/prod.list > /tmp/microsoft-prod.list || exit $ERR_MOBY_APT_LIST_TIMEOUT
-        fi
```
Comment (artunduman) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:74]: If I'm not missing anything, seems like this branch will never be executed

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 2]
```diff
@@ -66,16 +66,7 @@ installDeps() {
 }
 
 updateAptWithMicrosoftPkg() {
-    if [[ $(isARM64) == 1 ]]; then
-        if [[ "${UBUNTU_RELEASE}" == "22.04" || "${UBUNTU_RELEASE}" == "24.04" ]]; then
-            retrycmd_if_failure_no_stats 120 5 25 curl https://packages.microsoft.com/config/ubuntu/${UBUNTU_RELEASE}/prod.list > /tmp/microsoft-prod.list || exit $ERR_MOBY_APT_LIST_TIMEOUT
```
Comment (artunduman) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:71]: Same cmd as line 76

File: pkg/agent/testdata/RawUbuntuContainerd/line62.sh [hunk 3]
```diff
@@ -60,16 +60,7 @@ installDeps() {
 }
 
 updateAptWithMicrosoftPkg() {
-    if [[ $(isARM64) == 1 ]]; then
```
Comment (chengliangli0918) [inline pkg/agent/testdata/RawUbuntuContainerd/line62.sh:63]: where is corresponding change in for pkg/templates/templates_generated.go?

Comment (artunduman) [reply pkg/agent/testdata/RawUbuntuContainerd/line62.sh:63]: I'm wondering that myself too
@anujmaheshwari1 @cameronmeissner Do you know if there is a separate command to run, than `make generate`

**Review by chengliangli0918 — APPROVED** [review]

File: pkg/agent/testdata/RawUbuntuContainerd/line62.sh (hunk 3 above)
Comment (artunduman) [reply pkg/agent/testdata/RawUbuntuContainerd/line62.sh:63]: Seems expected, discussed offline (and in [this PR](https://github.com/Azure/AgentBaker/pull/4434))

**Review by djsly — APPROVED** [review]
Logic looks right, I didn't check is there is a possibility that Arm64 and UBUNTU_RELEASE -- 18.04 would be a valid condition when calling `updateAptWithMicrosoftPkg`

**Pull request merged**